src/test/java/com/softserve/academy/spaced/repetition/controller/DeckControllerTest.java

```diff
@@ -0,0 +1,217 @@
+                .build();
+    }
+
+    @Test
+    public void getDecksById() throws Exception {
+        when(deckService.getDeck(eq(4L))).thenReturn(createDeck());
+        mockMvc.perform(get("/api/admin/decks/{deckId}", 4L)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\"name\": \"Java interview #4\"," +
+                        "  \"owner\": \"admin@gmail.com\"," +
+                        "  \"description\": \"Part 4\"," +
+                        "  \"deckId\": 4," +
+                        "  \"rating\": 0," +
+                        "  \"category\": \"Java\"," +
+                        "  \"categoryId\": 1," +
+                        "  \"links\": [{" +
+                        "        \"rel\":\"self\"," +
+                        "          \"href\":\"http://localhost/api/admin/decks/4\"" +
+                        "      }]" +
+                        "}"));
+    }
+
+    private Deck createDeck() {
+        Deck deck = createDeck(4L, "Java interview #4", "Part 4",
+                "admin@gmail.com",0,1L,"Java");
+        return deck;
+    }
+
+    @Test
+    public void getDecksByPage() throws Exception {
+        when(deckService.getPageWithAllAdminDecks(NUMBER_PAGE, ADMIN_DECKS_SORT_BY, ADMIN_DECKS_ASCENDING)).thenReturn(createDecks());
+        mockMvc.perform(get("/api/admin/decks?p="+NUMBER_PAGE+"&sortBy="+ ADMIN_DECKS_SORT_BY +"&asc="+ ADMIN_DECKS_ASCENDING)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size", is(QUANTITY_ADMIN_DECKS_IN_PAGE)))
+                .andExpect(jsonPath("$.sort[:1].ascending", hasItem(ADMIN_DECKS_ASCENDING)))
+                .andExpect(jsonPath("$.sort[:1].descending", hasItem(!ADMIN_DECKS_ASCENDING)))
+                .andExpect(jsonPath("$.sort[:1].property", hasItem(ADMIN_DECKS_SORT_BY)));
+    }
+
+    public static Deck createDeck(long idDeck,String nameDeck, String description,
+                                  String accountEmail, double rating, long idCategory,
+                                  String categoryName) {
+        Deck deck = new Deck();
+        deck.setName(nameDeck);
+        User user = new User();
+        Account account = new Account();
+        account.setEmail(accountEmail);
+        user.setAccount(account);
+        deck.setDeckOwner(user);
+        deck.setDescription(description);
+        deck.setId(idDeck);
+        deck.setRating(rating);
+        Category category = new Category();
+        category.setName(categoryName);
+        category.setId(idCategory);
+        deck.setCategory(category);
+        return deck;
+    }
+
+    private Page<Deck> createDecks() throws ParseException {
+        List<Deck> deckList = new ArrayList<>();
+
```

Comment:
You can create a loop from 1 to 10 not to repeat.

Reply:
I think this is impossible, because besides the numbers we have other String parameters, with the rest I agree I will correct it.

Reply:
This is a test data. Next course name is acceptable:
`String course = "Java interview #" + i;`

Any way you are not doing any asserts on this data.